consensus/consensus-types/src/common.rs

```diff
@@ -219,28 +219,25 @@ pub enum Payload {
 }
 
 impl Payload {
-    pub fn transform_to_quorum_store_v2(
+    pub fn transform_to_payload_with_limits(
         self,
         max_txns_to_execute: Option<u64>,
         block_gas_limit_override: Option<u64>,
     ) -> Self {
         match self {
-            Payload::InQuorumStore(proof_with_status) => Payload::InQuorumStoreWithLimit(
-                ProofWithDataWithTxnLimit::new(proof_with_status, max_txns_to_execute),
-            ),
+            Payload::DirectMempool(_) => {
+                panic!("Payload is in direct mempool format");
```

Comment:
Can a malicious validator send a payload with direct mempool and crash other honest validators?

Reply:
I pushed a change to always check the payload is not direct mempool before calling this function as it's called from the proposal generator.

In general, the panics/unreachables throughout should be prevented by the verify on receiving the messages
https://github.com/aptos-labs/aptos-core/blob/c0a759b71b73aa20e8c16286cec9e30eea2871d3/consensus/consensus-types/src/common.rs#L513